partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -127,26 +145,104 @@ internal object RexConverter {
             throw IllegalArgumentException("unsupported rex $node")
 
         override fun visitExprLit(node: ExprLit, context: Env): Rex {
-            val type = when (val value = node.value) {
-                // Specifically handle the lack of an infinite precision/scale
-                // TODO: PartiQLValue won't be in AST soon
-                is DecimalValue -> {
-                    when (val decimal = value.value) {
-                        null -> PType.decimal()
-                        else -> PType.decimal(decimal.precision(), decimal.scale())
-                    }
-                }
-                else -> value.type.toPType()
-            }
             val cType = CompilerType(
-                _delegate = type,
-                isNullValue = node.value.isNull,
-                isMissingValue = node.value is MissingValue
+                _delegate = node.lit.toPType(),
+                isNullValue = node.lit is LiteralNull,
+                isMissingValue = node.lit is LiteralMissing
             )
-            val op = rexOpLit(node.value)
+            val op = rexOpLit(node.lit.toPartiQLValue())
             return rex(cType, op)
         }
 
+        private fun Literal.toPType(): PType {
+            return when (this) {
+                is LiteralNull, is LiteralMissing -> PType.unknown()
+                is LiteralString -> PType.string()
+                is LiteralBool -> PType.bool()
+                is LiteralDecimal -> {
+                    if (this.value.scale() == 0) {
+                        PType.numeric()
```

Comment:
TODO Alan. since we're keeping SQL's `numeric`, need to
- look into modeling of `PType.numeric`, `Datum.numeric()`, and `PartiQLValue.intValue` further
- depending on the AST modeling of literal numbers, may also add a numeric

Reply:
Checked w/ John -- the behavior of PType.numeric will get changed in some upcoming changes next week. Added a todo for now.